src/chapters/loops/for-loops.rst

```diff
@@ -0,0 +1,90 @@
+
+::
+
+   for (initial expression; loop condition; update expression) {
+      loop body
+   }
+
+Notice that in the first line, within parentheses, the components **initial expression**, **loop condition**, and **update expression** are separated by semicolons. Let's look at these components in detail.
+
+- The statement ``let i = 0`` is executed exactly once, at the *beginning* of loop execution. The variable ``i`` is the **loop variable**. 
+- The boolean expression ``i < 51`` is the **loop condition**. This condition is evaluated before each loop iteration, or repetition. If the condition is ``true`` then the loop executes again. If the condition is ``false`` then the loop ceases execution and the program continues to execute the code below the loop. In the example above, an equivalent condition would be ``i <= 50``. It is common to use ``<`` when writing loop conditions, so that the last value used in the loop is *one less than* then right-hand side of the ``<`` symbol.
+- The statement ``i++`` is the **update expression**. This expression is executed at the *end* of each loop iteration.
+- The block of code surrounded with brackets (``{ }``) is the **loop body**. The loop body is executed once for each iteration of the loop.
+
+Flow of Execution of the **for** Loop
+-------------------------------------
+
+In just a few lines of code, a ``for`` loop contains a lot of detailed logic, so let's spend some time breaking down the flow of execution for the particular loop that we've been looking at.
+
+.. sourcecode:: js
+
+   for (let i = 0; i < 51; i++) {
+      console.log(i);
+   }
+
+Here is a step-by-step description of how this loop executes:
+
+#. When the program reaches the ``for`` loop, the initial expression ``let i = 0`` is executed, declaring the variable ``i`` and initializing it to the value ``0``.
+#. The loop condition ``i < 51`` is evaluated, returning ``true`` becuase 0 is less than 51. 
+#. Since the condition is true, the loop body executes, printing 0.
+#. After the execution of the loop body, the update expression ``i++`` is executed, setting ``i`` to 1. This completes the first iteration of the loop.
+#. Steps 2 through 4 are repeated, using the new value of ``i``. This continues until the loop condition evaluates to ``false`` in step 2, ending the loop. In this example, this occurs when ``i < 51`` is ``false`` for the first time. Since our update expression adds 1 after each iteration, this occurs when ``i`` is 51 (so ``51 < 51`` is ``false``). At that point, the loop body will have executed exactly 51 times, with ``i`` having the values 0...50.
+
+In general, we can visualize the flow of execution of a ``for`` loop as a flowchart.
+
+.. figure:: figures/for-loop-flow.png
+   :height: 700px
+   
+   Flow of execution of a ``for`` loop
+
+.. warning:: If the body of a ``for`` loop has only a single line of code, then it may be placed on the line below the ``for`` statement *without* enclosing brackets, like this:
```

Comment:
Eliminate the warning box!  If we want to encourage the students to ALWAYS use {} in for loops, then we should not give them an out.  We want to avoid contradicting our own style guide.  Perhaps we can add the 1-line-for-loop exception in an appendix, but let’s keep it out of the main content.

Reply:
Valid point. The thing I'm guarding against here is the instructor or TA who tells a student, "You don't really need those brackets all the time." I fear that if we bury this comment in an appendix, then this scenario will happen more readily.

Reply:
On the other hand, how often do you code single line for loops?  I'd let the TA who says, "You don't always need {}," explain the details.